### ISSUE TITLE
CupertinoSwipeBox is broken

### ISSUE THREAD

**Issue by schott12521** [alexzhirkevich/compose-cupertino#72]
This is blocking the release of compose-cupertino 0.2.0-alpha05 which will enable support for compose-multiplatform 1.7.0

I've got one PR to breakdown the components to help with debugging: https://github.com/alexzhirkevich/compose-cupertino/pull/68

But I'm pretty close to a new PR that basically rewrites the SwipeBox from the ground up using a simplified approach with AnchorDraggable.

There's a known bug with compose-multiplatform 1.7.0 and anchoredDraggable with overscroll that calculates 2D gestures instead of 1D, which prevents us from using overscroll for now: https://android-review.googlesource.com/c/platform/frameworks/support/+/3284436

**Comment by schott12521** [comment]
For some additional references, I'm pretty sure the original author's intention was to mimic iOS's `.swipeActions(edge:allowsfullswipe:content:)`: https://developer.apple.com/documentation/swiftui/view/swipeactions(edge:allowsfullswipe:content:)

And for my implementation, I've been taking a lot of inspiration from compose-swipebox, which gets pretty close to iOS like feeling with a few things notably missing: https://github.com/KevinnZou/compose-swipeBox

**Comment by schott12521** [comment]
Here's my current implementation, I'll upload a video later:

https://gist.github.com/schott12521/5b5d7ce91355a72883ec97c472211a63

Things I have working:

- Single action item working on Swipe from Start to End, with an "expanded" state, and a "fully expanded" that vibrates to indicate the action should be taken
- Horizontal overscroll works, but its a bit janky because the action item takes up the full width of the parent, so you can see the tint when overscrolling past the start. This was done be reworking the CupertinoOverscrollEffect to only apply for X offsets, this is temporary and should be replaced with the compose-multiplatform version when its released

Things that need to be added back in / fixed:

- Ability to toggle full expansion
- Multiple Action Items, where full expansion will make the first/last item be taken
- Animations to indicate the action will be taken on full expansion
- Migrate to compose-multiplatform overscroll once it is fixed

**Comment by schott12521** [comment]
https://github.com/user-attachments/assets/ee350503-f1ae-4900-99eb-d3b791f3d380

Pretty happy with this implementation, there's a lot to be done to have full feature parity with the iOS version, but I think this is worth releasing / unblocking the build. I'll get a PR out shortly.

**Comment by schott12521** [comment]
I've submitted https://github.com/alexzhirkevich/compose-cupertino/pull/74 to fix this
